### ISSUE TITLE
Haddock build failure with GHC 8.2

### ISSUE THREAD

**Issue by snoyberg** [goldfirere/singletons#242]
```
src/Data/Singletons/Prelude/Function.hs:107:3: error:
    Haddock documentation not (yet) handled by Template Haskell
    |
107 |   -- | '&' is a reverse application operator.  This provides notational
    |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
Base.hs:55)
```

**Comment by RyanGlScott** [comment]
This was fixed upstream in #186. I'll look into cutting another minor release soon with the minor Haddock fixes from that patch.

**Comment by RyanGlScott** [comment]
I've uploaded [`singletons-2.3.1`](http://hackage.haskell.org/package/singletons-2.3.1) to Hackage with a fix.
